src/random_test.cpp

```diff
@@ -1709,7 +1709,9 @@ void load_default_data(Table *table, Thd1 *thd) {
   thd->ddl_query = false;
   static int initial_records =
       options->at(Option::INITIAL_RECORDS_IN_TABLE)->getInt();
-  int rec = rand_int(initial_records);
+  int rec = options->at(Option::EXACT_INITIAL_RECORDS)->getBool()
+                ? initial_records
+                : rand_int(initial_records);
   for (int i = 0; i < rec; i++) {
     table->InsertRandomRow(thd, false);
```

Comment:
The inserted number of rows in the table does not match the provided number of records. For eg
./pstress-ps --tables 1 --logdir=`pwd`/log --threads 10 --socket /tmp/mysql_22000.sock --seconds 1 --records 20 --exact-initial-records
This inserts 
mysql> select count(*) from tt_1_p;
+----------+
| count(*) |
+----------+
|       27 |
+----------+
1 row in set (0.00 sec)

mysql> select count(*) from tt_1;
+----------+
| count(*) |
+----------+
|       35 |
+----------+
1 row in set (0.01 sec)

Reply:
seconds 1 mean load is executed for 1 seconds.  try with --seconds 0. there may be 10% less becuase of some failure of DML due to unique/primary index but not huge difference

Reply:
mohit.joshi@qaserver-06:~/pstress_8.0/src$ ./pstress-ps --tables 1 --logdir=`pwd`/log --threads 10 --socket /tmp/mysql_22000.sock --seconds 0 --records 20 --exact-initial-records
- Connecting to default.node.tld []...
metadata created randomly
The value of rec:20
The value of rec:20
starting random load in 10 threads.
Saving metadata to file /home/mohit.joshi/pstress_8.0/src/log/step_1.dll
COMPLTED
mysql> select count(*) from tt_1;
+----------+
| count(*) |
+----------+
|       20 |
+----------+
1 row in set (0.00 sec)

mysql> select count(*) from tt_1_p;
+----------+
| count(*) |
+----------+
|       15 |
+----------+
1 row in set (0.00 sec)

Reply:
For --tables 1 --records 200, --exact-initial-records
select count(*) from tt_1_p;    80
select count(*) from tt_1;   199

Reply:
tt_1_p is list partition. that's fine. we can't expect lot of enties in them.

Reply:
Error Table has no partition for value 66.

Reply:
Ok